pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
@sdemos @euank I don't have further changes. I think we want to keep the sleep, but have it be interruptible. #93 is still an issue with or without this change, it requires a future addition to trap the shutdown and close the stop channel to correctly close all routines/sleeps. Resolved to satisfaction?